src/App/ComplexGeoData.cpp

```diff
@@ -386,4 +389,208 @@ char ComplexGeoData::elementType(const char *name) const {
     return 0;
 }
 
+void ComplexGeoData::setPersistenceFileName(const char *filename) const {
+    if(!filename)
+        filename = "";
+    _PersistenceName = filename;
+}
+
+void ComplexGeoData::Save(Base::Writer &writer) const {
+
+    if(!getElementMapSize()) {
+        writer.Stream() << writer.ind() << "<ElementMap/>\n";

```

Comment:
why are we writing elementmap tags in complexgeodata? should this logic be in elementmap instead?

Reply:
Yes, I agree, that would be a good refactor.